### PR TITLE
azurerm_application_gateway  - Support for Managed Identity (resolves #3640)

### LINKED ISSUE #3640: Managed Identity for Application Gateway
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support the Managed Service Identity for Application Gateway.

### New or Affected Resource(s)

* azurerm_application_gateway

### Potential Terraform Configuration

```hcl
resource "azurerm_application_gateway" "network" {
    ...
    identity = "${azurerm_user_assigned_identity.test.id}"
    ...
}
```

### References

* https://godoc.org/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-02-01/network#ApplicationGateway

* #0000

### PR DESCRIPTION
**Pull request by fxgsell**
resolves #3640

### CODE REVIEW

File: azurerm/resource_arm_application_gateway.go [hunk 1]
```diff
@@ -50,6 +50,40 @@ func resourceArmApplicationGateway() *schema.Resource {
...
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_application_gateway.go]: all new fields in the provider are case-sensitive, as such can we remove this?

```suggestion
```

File: azurerm/resource_arm_application_gateway.go [hunk 2]
```diff
@@ -50,6 +50,40 @@ func resourceArmApplicationGateway() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"identity": {
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_application_gateway.go]: could we [add this new block to the documentation](https://github.com/terraform-providers/terraform-provider-azurerm/blame/master/website/docs/r/application_gateway.html.markdown#L171)?

File: azurerm/resource_arm_application_gateway.go (hunk 1 above)
Comment (fxgsell) [reply azurerm/resource_arm_application_gateway.go]: OK

File: azurerm/resource_arm_application_gateway.go (hunk 2 above)
Comment (fxgsell) [reply azurerm/resource_arm_application_gateway.go]: OK

**Comment by fxgsell** [comment]
Thanks @tombuildsstuff ! 

Also I've chosen to create an identity block rather than a simple string, for 3 reasons:

- Followed the way the SDK does.
- Followed the way identities are handled in other resources.
- Future proof it (in case support for System identities and multiple identities is added later).

That seems reasonable?

File: azurerm/resource_arm_application_gateway.go [hunk 3]
```diff
@@ -49,6 +49,33 @@ func resourceArmApplicationGateway() *schema.Resource {
...
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (katbyte) [inline azurerm/resource_arm_application_gateway.go]: Would it make sense to mark this as `Optional` with a `Default`

Comment (fxgsell) [reply azurerm/resource_arm_application_gateway.go]: Hi @katbyte , yep that makes sens, done. Also fixed the acc tests.

**Comment by dbilleci-lightstream** [comment]
can we get a bump on this? in need of identity on AG for use with keyvault certificates on HTTPS listener

**Comment by katbyte** [comment]
Thanks for the revisions @fxgsell, tests pass and LGTM now 👍 

(seems this didn't go through before i merged)

**Pull request merged**